src/main/assets/js/courts-table-search.ts

```diff
@@ -0,0 +1,249 @@
+   * is used, otherwise it is updated on each search toggle or courts search
+   *
+   * @param filterName Either 'name' or 'date' depending on the type of toggle
+   * @param defaultValue set to false by default on initial load
+   */
+  static setUpTable(filterName: string, defaultValue = false): void {
+    defaultValue
+      ? CourtsTableSearch.setUpTableData(
+        $(this.searchCourtsFilterId).val() as string, // if the user clicks 'back' in the browser after view or editing
+        false, orderToggleState.ASC, orderToggleState.INACTIVE)
+      : CourtsTableSearch.setUpTableData(
+        $(this.searchCourtsFilterId).val() as string,
+        $(`#main-content input[name=${this.toggleClosedCourtsDisplay}]`).prop('checked'),
+        $(this.courtsNameAscToggleId).val() as string,
+        $(this.courtsUpdatedAscToggleId).val() as string);
+
+    filterName == 'name' ?
+      CourtsTableSearch.switchTableClasses($(this.courtsNameAscToggleId),
+        $(this.tableCourtsNameId), $(this.tableCourtsUpdatedId)) :
+      CourtsTableSearch.switchTableClasses($(this.courtsUpdatedAscToggleId),
+        $(this.tableCourtsUpdatedId), $(this.tableCourtsNameId));
+  }
+
+  /**
+   *
+   * Retrieves the filtered results from the toggles and search, and sets the html
+   * of the tbody element of the table accordingly
+   *
+   * @param searchFilterValue the search value from the 'search courts or tribunals' input field
+   * @param includeClosedCourts whether or not to include closed courts in the result
+   * @param orderNameAscendingFilter whether to sort by name asc or desc
+   * @param orderUpdatedAscendingFilter whether to sort by last updated date asc or desc
+   * @private
+   */
+  private static setUpTableData(searchFilterValue: string, includeClosedCourts: boolean,
+    orderNameAscendingFilter: string, orderUpdatedAscendingFilter: string): void {
+    const filteredCourts = CourtsTableSearch.filterCourts(this.getExistingTableData(),
+      searchFilterValue, includeClosedCourts,
+      orderNameAscendingFilter, orderUpdatedAscendingFilter);
+    $(this.courtsResultsSection).html(CourtsTableSearch.getCourtsTableBody(filteredCourts));
+    searchFilterValue.length
+      ? $(this.numberOfCourts).show().text('Showing '
+        + filteredCourts.filter(d => d.visible).length + ' results')
+      : $(this.numberOfCourts).hide();
+  }
+
+  /**
+   *
+   * Get the court data by extracting it from the table
+   *
+   * @private
+   */
+  private static getExistingTableData(): CourtItem[] {
+    const courtItems = [] as CourtItem[];
+    $(this.tableData).find('tr').each(function(i, row) {
+      if (i == 0)
+        // the first index is for the column names; i.e name, displayed, updated_at
+        return;
+
+      const courtItem = {} as CourtItem;
+      $(row).find('td').each(function(i, dataCell) {
+
+        switch ($(dataCell).attr('type')) {
+          case 'name':
+            courtItem.name = $(dataCell).attr('value');
```

Comment:
Do we need to use attributes for getting the value of the td elements?

Reply:
wouldn't it be better to potentially, on a row by row basis, have data-attributes for name, closed, last-updated on a court? This removes the need to parse down into each individual row

```
<tr class="govuk-table__row" data-court-name="Aberdeen Tribunal Hearing Centre" data-court-updated-at="05 Mar 2021" data-court-closed="false">
        <td scope="row" class="govuk-table__cell">Aberdeen Tribunal Hearing Centre</td>
        <td scope="row" class="govuk-table__cell">05 Mar 2021</td>
        ...
</tr>
```

Reply:
The GDS design component only allows attributes on td elements, but happy to look into setting data for each one instead, or to keep it as is. I know Andrew and I discussed this on a call the other day